### PR TITLE
Tuning for GEMM: optional alpha beta

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
Some tuning for GEMM, e.g. no alpha beta if not needed; some cleanup too

Signed-off-by: Alexandre Eichenberger <alexe@us.ibm.com>

### CODE REVIEW

File: src/Conversion/KrnlToAffine/KrnlToAffine.cpp [hunk 1]
```diff
@@ -1083,15 +1083,13 @@ class KrnlCopyToBufferLowering : public OpRewritePattern<KrnlCopyToBufferOp> {
       }
     }
     SmallVector<Value, 4> loopIndices;
-    genCopyLoops(buffMemref, sourceMemref, srcIndexMap, srcLoopMap, padVal,
-        zero, starts, bufferReadUBs, bufferPadUBs, loopIndices, 0, buffRank,
-        false);
+    genCopyLoops(buffMemref, sourceMemref, srcLoopMap, padVal, zero, starts,
```
Comment (AlexandreEichenberger) [inline src/Conversion/KrnlToAffine/KrnlToAffine.cpp:1086]: Cleanup, one param was not needed.

File: src/Conversion/ONNXToKrnl/Math/Gemm.cpp [hunk 2]
```diff
@@ -97,19 +97,19 @@ struct ONNXGemmOpLowering : public ConversionPattern {
     // Scope for krnl EDSC ops
     using namespace mlir::edsc;
     using namespace mlir::edsc::intrinsics;
-    // ScopedContext scope(rewriter, loc);
 
     // R is result (alloc).
     Value A(operandAdaptor.A()), B(operandAdaptor.B()), R(alloc);
-    MemRefBoundsCapture aBound(A), rBound(R);
+    MemRefBoundsIndexCapture aBound(A), rBound(R);
```
Comment (AlexandreEichenberger) [inline src/Conversion/ONNXToKrnl/Math/Gemm.cpp:103]: Use IndexExpr for iterates... it make it use constant, and that prevent an annoying warning from affine.

File: test/numerical/TestGemm.cpp [hunk 3]
```diff
@@ -175,21 +174,25 @@ int main(int argc, char *argv[]) {
...
   printf("RapidCheck test case generation.\n");
   rc::check("Gemm implementation correctness", []() {
-    const auto aTrans = *rc::gen::inRange(0, 2);
-    const auto bTrans = *rc::gen::inRange(0, 2);
     const auto I = *rc::gen::inRange(1, 50);
     const auto J = *rc::gen::inRange(1, 50);
     const auto K = *rc::gen::inRange(1, 50);
-
-    RC_ASSERT(isOMGemmTheSameAsNaiveImplFor(I, J, K, aTrans, bTrans));
+    const auto aTrans = *rc::gen::inRange(0, 2);
```
Comment (AlexandreEichenberger) [inline test/numerical/TestGemm.cpp:180]: Added more tests to cover new opts

**Review by caoimhinuibrian — APPROVED** [review]
LGTM

**Pull request merged**